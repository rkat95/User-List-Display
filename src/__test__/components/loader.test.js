import { render, screen, cleanup } from "@testing-library/react";
import React from "react";
import Loader from "../../components/loader/loader";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("Loader tests", () => {
  it("should render loader", () => {
    render(<Loader />);
    const heading = screen.getByTestId("loader");
    expect(heading).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<Loader />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
