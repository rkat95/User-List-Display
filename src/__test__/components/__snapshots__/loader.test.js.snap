// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Display tests should match snapshot 1`] = `
<div
  className="loader-wrapper"
  data-testid="loader"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary loader-circle css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;

exports[`Loader tests should match snapshot 1`] = `
<div
  className="loader-wrapper"
  data-testid="loader"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary loader-circle css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;
